modules/Bio/EnsEMBL/Variation/DBSQL/AnnotationFileAdaptor.pm

```diff
@@ -0,0 +1,280 @@
+    local $/ = undef;
+    my $json_string = <IN>;
+    close IN;
+    
+    # parse JSON into hashref $config
+    $config = JSON->new->decode($json_string) or throw("ERROR: Failed to parse config file $config_file");
+  }
+  
+  $self->{config} = $config;
+  $self->db->annotation_config($config) if $self->db;
+
+  ## set up root dir
+  my $root_dir = '';
+  if($ENV{ENSEMBL_VARIATION_ANNOTATION_ROOT_DIR}) {
+    $root_dir = $ENV{ENSEMBL_VARIATION_ANNOTATION_ROOT_DIR}.'/';
+  }
+  elsif($self->db && $self->db->annotation_root_dir) {
+    $root_dir = $self->db->annotation_root_dir.'/';
+  }
+  
+  $self->{root_dir} = $root_dir;
+
+  my $tmpdir = cwd();
+  if($ENV{ENSEMBL_VARIATION_ANNOTATION_TMP_DIR}) {
+    $tmpdir = $ENV{ENSEMBL_VARIATION_ANNOTATION_TMP_DIR}.'/';
+  }
+  elsif($self->db && $self->db->annotation_tmp_dir) {
+    $tmpdir = $self->db->annotation_tmp_dir.'/';
+  }
+
+  $self->{tmpdir} = $tmpdir;
+
+  throw("ERROR: No annotation data defined in config file") unless $config->{annotation_type} && keys %{$config->{annotation_type}};
+ 
+  if ($self->db) {
+    $self->{species} = lc($self->db->species);
+    $self->{assembly} = lc($self->db->dnadb->get_CoordSystemAdaptor->fetch_all->[0]->version);
+  }
+
+  return $self;
+}
+
+=head2 fetch_by_annotation_type
+  Example    : my $gerp_file = $annotation_file_adaptor->fetch_by_annotation_type('gerp');
+               my $cadd_file = $annotation_file_adaptor->fetch_by_annotation_type('cadd');
+  Description: Fetches AnnotationFile by given annotation type
+  Returntype : Bio::EnsEMBL::Variation::GERPFile or Bio::EnsEMBL::Variation::CADDFile
+  Exceptions : Throws on wrong annotation type
+  Caller     : general
+  Status     : Stable
+=cut
+
+sub fetch_by_annotation_type {
+  my $self = shift;
+  my $type = shift;
+
+  my $annotation_hash = $self->_get_annotation_hash($type);
+
+  if ($type eq 'gerp') {
+    return $self->_get_gerp_annotation_file($annotation_hash);
+  } elsif ($type eq 'cadd') {
+    return $self->_get_cadd_annotation_file($annotation_hash);
+  } else {
+    die("Unknown annotation type. Annotation type must be cadd or gerp.");
+  }
+}
+
+sub _get_annotation_hash {
+  my $self = shift;
+  my $annotation_type = shift;
+  my $config = $self->{config};
+  my $species = $self->{species};
+  my $assembly = $self->{assembly};
+  my @annotations  = grep {lc $_->{species} eq $species && lc $_->{assembly} eq $assembly} @{$config->{annotation_type}->{$annotation_type}};
+  if (scalar @annotations == 1) {
+    return $annotations[0];
+  } elsif (scalar @annotations > 1) {
+    die "Found more than one annotation file for annotation type $annotation_type for species $species and assembly $assembly\n";
+  } else {
+    die "Could not find annotation file for annotation type $annotation_type for species $species and assembly $assembly\n";
+  }
+}
+
+sub _get_gerp_annotation_file {
+  my $self = shift;
+  my $hash = shift;
+  return Bio::EnsEMBL::Variation::GERPFile->new(
```

Comment:
Method  `_get_gerp_annotation_file` quite similar to `_get_cadd_annotation_file`.
Maybe we could use a more generic method using `GERPFile` or `CADDFile` depending on the type in the parameter